src/pipeline.jl

```diff
@@ -1,25 +1,56 @@
 ## Stubs
 function _recipe_init!(plt, plotattributes, args) end
 function _recipe_after_user!(plt, plotattributes, args) end
-function _recipe_after_plot!(plt, plotattributes, args) end
-function _recipe_after_series!(plt, plotattributes, args, series_ind) end
+function _recipe_after_plot!(plt, plotattributes, kw_list) end
+function _recipe_before_series!(plt, kw, kw_list) end
 function _recipe_finish!(plt, plotattributes, args) end
+
 ##
 # Here comes the specification of when which recipe is processed.
 # It contains functions before and after every stage for interaction with the plotting package.
 
-function recipe_pipeline!(plt,             # frontend specific representation of a plot
-                         plotattributes,  # current state of recipe keywords
-                         args,            # set of arguments passed by the user
-  )
-  _recipe_init!(plt, plotattributes, args)
-  kw_list = _process_userrecipes(plt, plotattributes, args)
-  _recipe_after_user!(plt, plotattributes, args)
-  kw_list = _process_plotrecipes(plt, plotattributes, args)
-  _recipe_after_plot!(plt, plotattributes, args)
-  for (series_ind, series) in enumerate(series_list)
-    kw_list = _process_seriesrecipe(plt, plotattributes)
-    _recipe_after_series!(plt, plotattributes, series_ind)
-  end
-  _recipe_finish!((plt, plotattributes, args)
+function recipe_pipeline!(plt,              # frontend specific representation of a plot
+                         plotattributes,    # current state of recipe keywords
+                         args;              # set of arguments passed by the user
+                         type_aliases)
```

Comment:
I'd say either we give it a default value as kwarg or we make it a positional argument, no? I'd rather go for a kwarg with default.

Reply:
The reason for avoiding the default is that it forces the user to provide a value and I was not sure if an empty `Dict` was a safe default in the sense that it would not lead to strange behavior (if someone forgets to provide it).

Reply:
An empty `Dict` is fine.